spec/support/i18n_cov.rb

```diff
@@ -73,6 +74,15 @@ def print_message(all, used)
     puts "Coverage report generated for I18n to #{Dir.pwd}/coverage/i18n. " +
       "#{used.count} / #{all.count} keys (#{percent}%) covered."
   end
+
+  def check_covered(all, used)
+    percent = ((used.count.to_f / all.count) * 100).round(2)
+    return if percent == 100
+
+    warn "I18n coverage (#{percent}%) is below the expected minimum coverage (100%)"
+    warn "I18nCov failed with exit 1"
+    Kernel.exit 1
```

Comment:
I don't think this is the correct place to exit with 1 and raise an alert. It looks like this will conflate the test results with the coverage results. It'd show for instance in Jenkins that your tests have failed when actually it would have been this coverage. 

I'd suggest that you have a stage defined in https://github.com/alphagov/content-publisher/blob/8a9582c738b62181a3d17bc56180a505b9f4cd93/Jenkinsfile#L23-L27 which checks whether the file of unused keys is empty or not and then fail the build accordingly if necessary.

Reply:
@kevindew I'm struggling a bit to see your point of view here. Reading the file is a good idea if we do decide to go this way, as otherwise we'd have to run the tests again.

I'm basing this on the way [SimpleCov would behave](https://github.com/colszowka/simplecov#getting-started) if we set `SimpleCov.minimum_coverage 100` (although it would fail with an exit code of `2`). Are we saying we would do something special for SimpleCov as well? It's also not related to any test failures.

Reply:
Yeah, I wouldn't expect to run tests again

In my experiences of running these things in CI is you use running the tests to generate the information and the CI machine itself is configured to read this information and make decisions on the build.

For example for Simplecov and Jenkins rather than configuring it in Ruby you'd more conventionally hook the report into the [code coverage API and specify that to fail]((https://cdn.jsdelivr.net/gh/jenkinsci/code-coverage-api-plugin@master/images/config.png))